sdk/src/claim.rs

```diff
@@ -2522,41 +2522,63 @@ impl Claim {
 
                             if let Some(ingredient_label) = manifest_label_from_uri(&redacted_uri) {
                                 // can we find a reference in the ingredient list
-                                if let Some(ingredient) = svi.manifest_map.get(&ingredient_label) {
-                                    // does the assertion exist
-                                    if let Some(readaction_label) =
+                                if let Some(ingredient_claim) =
+                                    svi.manifest_map.get(&ingredient_label)
+                                {
+                                    // The referenced manifest exists, so far so good.
+                                    // now get the assertion label and try to resolve it.
+                                    if let Some(redaction_label) =
                                         assertion_label_from_uri(&redacted_uri)
                                     {
-                                        let (label, instance) =
-                                            Claim::assertion_label_from_link(&readaction_label);
-                                        parent_tested = Some(
-                                            ingredient.get_assertion(&label, instance).is_some(),
-                                        );
-                                    } else {
-                                        parent_tested = Some(false);
+                                        if ingredient_claim
+                                            .assertion_hashed_uri_from_label(&redaction_label)
+                                            .is_some()
+                                        {
+                                            // The url reference is valid, now check if it was actually redacted
+                                            parent_tested = Some(false);
+                                            // Now if the assertion is not in the assertion store we are ok.
+                                            // Todo: would a zeroed out assertion show up here? if so we need to do a zero check
+                                            if ingredient_claim
+                                                .get_claim_assertion(&redaction_label, 0)
```

Comment:
I left this comment in the other PR. But the test is wether the Claim has an assertion in its assertion list (gathered, created, or 1.x lists).  It should/must be there.  Now what should fail is the second part that you have done here that is no matching assertion in the assertion store.  0s are already handled for you at that level.